Store and show notifications in a more robust way

jackcasey opened the pull request:
Store all messages with their own timeouts, build the message each time one is added or expires. Show each event in its own container and stack them up. Danger messages must be closed manually as before.

![alerts](https://user-images.githubusercontent.com/45890/27440021-06b77360-579c-11e7-97fb-ff6f2236341f.gif)

Addresses #473 

I'm not sure how to test the ENS to prove this actually fixes the bug. But this should be a much more robust notification system.

Known issues (To be dealt with in separate style pass):

- At mobile widths the stacking (fixed at 55px) means the alerts overlap each other as they expand vertically.
- When the alerts have enough text to wrap you also get the vertical overlap

Commit:
Store and show notifications in a more robust way

tayvano commented:
I have an absolute buttload of reveals tomorrow from testing bidding a few days ago. I'll give it a whirl! 

You have no idea how excited I am for this. Thank you in advance. 😁😁😁😁

Commit:
Fixed some indentation style

jackcasey commented:
Great. I dug through the ENS code to see if I could see anything funky but it only calls `notifier.warning`, `notifier.success` etc as far as I can see; so... :crossed_fingers:
If it works and you're happy with the PR style wise feel free to ping the bounty here: 0xea3344493c5da885ed29579f6f432e590166e031 :)

crptm commented:
I would prefer proper stacking, instead of level escalation.

jackcasey commented:
Can also do that, I was just replicating the current style where there is
only one UI element for all messages.

On Mon, 19 Jun. 2017, 6:44 pm crptm, <notifications@github.com> wrote:

> I would prefer proper stacking, instead of level escalation.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/kvhnuke/etherwallet/pull/591#issuecomment-309404293>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AACzQjUxabaeLm4Tt8-PnK8pFL525hu2ks5sFlEBgaJpZM4N9f37>
> .
>

Commit:
Changed to a stack of alerts.

jackcasey commented:
Added the stacking. I'm busy over the weekend so unfortunately don't have time to look into the minor stacking issues on mobile or when text wraps until Monday. Might not be a trivial fix as each alert is position:fixed so they don't know about each other at all...

tayvano commented:
@jackcasey Thank you! No worries. If it's CSS, that's the *one* thing I can do really, really well. Been meaning to update it anyways because the padding changed a bit ago. Timing is perfect.

crptm commented:
@jackcasey can we align with https://github.com/MyEtherWallet/MyEtherWallet/pull/20 this should fix your problems i believe

tayvano commented:
@crptm I'm going to pull it as the issues he's seeing are on current site currently anyways. Just need to clean up CSS
